### ISSUE TITLE
Missing theme when upgrade is going on

### ISSUE THREAD

**Issue by labkode** [owncloud/core#17221]
When we trigger the upgrade procedure from version 7.0.1 to 8.0.2, the theme setting is overwritten.
The users will see the maintenance mode with OwnCloud default theme and not the branded one.

Is there any way to avoid this glitch ?

Cheers

**Comment by DeepDiver1975** [comment]
The theme is deactivated on purpose. A theme can cause issues during upgrade so explicit action is required by the admin to reenable the theme.

**Comment by moscicki** [comment]
What kind of issue may a these cause?

**Comment by RealRancor** [comment]
@moscicki See https://github.com/owncloud/core/issues/9822#issuecomment-49948302 for example

**Comment by DeepDiver1975** [comment]
> What kind of issue may a these cause?

Theme developers use as dirty tricks as any other (PHP) developer does (e.g. use private APIs which we removed) or in some cases we changed the basic layout files and did cause strange side effects.

We cannot protect the system against themes which do bad things - as a result we decided to disable themes on upgrade.

(The only failsafe approach with respect to theming would be to have fully PHP-less theming mechanism - but to be honest: we better focus on proper file sync)
